### PR TITLE
allow to keep indentation with `opts.preserve_indent = true`

### PR DESCRIPTION
**Pull request by yavorskiy**
First attempt on optionally preserving indentation. 
@zxqfox it has been long time since we had this conversation but please take a look if you get a chance

### CODE REVIEW

File: tests/parse.spec.js [hunk 1]
```diff
@@ -132,6 +132,28 @@ describe('Comment string parsing', function() {
...
+       * 
+       *       third line
+       */
+      var a;
+    }, {
+      preserve_indent : true
+    })[0])
+      .eql({
+        description : '  Description first line\n    second line\n\n      third line',
+        source      : '  Description first line\n    second line\n\n      third line',
```
Comment (zxqfox) [inline tests/parse.spec.js]: I don't really like this uncancellable trims in source, guess we can left source as close as possible to source code. Hope we can :)

Comment (yavorskiy) [inline tests/parse.spec.js]: good point, I tend to agree

Comment (yavorskiy) [inline tests/parse.spec.js]: @zxqfox the more I think on it the more I tend to make this behavior default and rather add `opts.trim_indent` option

Comment (zxqfox) [inline tests/parse.spec.js]: Agree. But it's a major change, isn't it?

Comment (yavorskiy) [inline tests/parse.spec.js]: @zxqfox So, I ended up having default `opts.trim = true`. I would rather have it `false` by default, but you are right, it's kinda breaking change. Anyway it allows to preserve lines, space on left and right

Comment (zxqfox) [inline tests/parse.spec.js]: Fair enough. Would be nice to have it as a default in the next major ;-)

**Pull request merged**